### PR TITLE
Add a `--site` option to wabur 'new' and 'init' modes

### PR DESCRIPTION
**Pull request by ohler55**
Added the `--site` option to `wabur init` which copies the content of the export directory to a `site` directory of a new project.

### CODE REVIEW

**Comment by ashmaroli** [comment]
Made some corrections. Please review them when possible..

**Comment by ashmaroli** [comment]
Note: Having the same information in two places ( `README.md` and `MORE.md` ) means future corrections / alterations will need to be made twice..

File: lib/wab/impl/init.rb [hunk 1]
```diff
@@ -25,13 +25,16 @@ def initialize(path, config)
         types      = config[:rest] || []
         config_dir = "#{path}/config"
         lib_dir    = "#{path}/lib"
+        init_site  = config[:site]
 
-        raise WAB::Error.new('At least one record type is required for new or init.') if (types.nil? || types.empty?) && !config[:site]
+        if (types.nil? || types.empty?) && !init_site
+          raise WAB::Error.new("At least one record type is required for 'new' or 'init'.")
+        end
```
Comment (ohler55) [inline lib/wab/impl/init.rb]: Funny, that is the way I had it but changed because I figured you would put in a PR to change it anyway.

**Comment by ohler55** [comment]
Yeah, I know. I'm not sure what the best way to deal with that. Trying to keep the README shorter and provide details that can be linked to. There is probably a better way. Do you have any ideas?

File: tutorial/lesson-1/MORE.md [hunk 2]
```diff
@@ -282,14 +282,20 @@ to different URL paths or routes.
...
+
+To customize these files easily, they can be generated at the current WAB
+workspace by running the `wabur init` command with a `--site` option.
 
 ```
 $ wabur init --site
 ```
 
+> Pro-tip: Alternatively, the `--site` option can be passed to `wabur new` as
+well, to generate these files alongwith the config files and UI Controller.
```
Comment (ohler55) [inline tutorial/lesson-1/MORE.md]: spelling: 'alongwith' should be 'along with'. Same in the MORE.md file.

File: lib/wab/impl/init.rb (hunk 1 above)
Comment (ashmaroli) [reply lib/wab/impl/init.rb]: not when the resulting line is too long.. (more than 80-90 chars) :smiley:

File: tutorial/lesson-1/MORE.md (hunk 2 above)
Comment (ashmaroli) [reply tutorial/lesson-1/MORE.md]: ah yes.. my bad.. :grinning:

**Comment by ashmaroli** [comment]
> There is probably a better way. Do you have any ideas?

Keep only the "extra" stuff in `MORE.md`?

**Comment by ohler55** [comment]
Maybe. I will be disjointed though. I thought about multiple smaller files with links back to the original. Maybe that is the way to go.

**Comment by ohler55** [comment]
If that makes sense a new PR would be best. Lesson-1 was only touched in this PR to get the `--site` option explained.

**Comment by ashmaroli** [comment]
We'll sort this out in a new PR.. I'll push a typo-fix commit now..

**Comment by ashmaroli** [comment]
Did you know that running `wabur init -s` would overwrite an existing `site` and `index.html` without a warning?

**Comment by ohler55** [comment]
It is not supposed to. That is a bug.

**Comment by ohler55** [comment]
Is -v is used the log message says it already exists.

**Comment by ohler55** [comment]
I checked it again and it did not over-write.

**Comment by ashmaroli** [comment]
Right.. I can't reproduce it anymore.. must've been a latency with my text-editor.. all is well then.. except for a teeny nit that wabur still outputs the success message when nothing was re-initialized anyways.. but I can live with it for now..

**Comment by ashmaroli** [comment]
This is good-to-go, from my side..

**Comment by ohler55** [comment]
I thought of adding a number of files added message but decided it could wait for some experience using the tool.

**Comment by ohler55** [comment]
I think you have to do a review and approve since I started the PR.

**Comment by ashmaroli** [comment]
> but decided it could wait

Agreed.. its a nice addition.. plus, we could consider outputting paths relative to `--base`
```
    Written  lib/ui_controller.rb
     Copied  site/assets/css/wab.css
```

**Comment by ohler55** [comment]
Thats a good idea as well. I'll make and issue for that.

**Pull request merged**